src/physics/gyro_average.cxx

```diff
@@ -28,84 +28,118 @@
  **************************************************************/
 
 #include <bout/mesh.hxx>
-#include <globals.hxx>
+#include <bout/sys/timer.hxx>
 #include <difops.hxx>
+#include <globals.hxx>
 #include <gyro_average.hxx>
 #include <invert_laplace.hxx>
 
-const Field3D gyroTaylor0(const Field3D &f, const Field3D &rho) {
+Field3D gyroTaylor0(const Field3D& f, const Field3D& rho) {
   return f + SQ(rho) * Delp2(f);
 }
 
-const Field3D gyroPade0(const Field3D &f, BoutReal rho, int flags) {
-  
-  Field2D a = 1.0;
-  Field2D d = -rho*rho;
-  
+Field3D gyroPade0(const Field3D& f, BoutReal rho, int flags) {
+  const Field2D a = 1.0;
+  const Field2D d = -rho * rho;
+
   // Invert, leaving boundaries unchanged
-  return invert_laplace(f, flags, &a, nullptr, &d);
+
+  Timer timer("invert");
+
+  auto* lap = Laplacian::defaultInstance();
```

Comment:
Can/should we allow an options section to be passed through should we want to use a non-default Laplacian setup?